Sources/SwiftFormatPrettyPrint/TokenStreamCreator.swift

```diff
@@ -67,14 +67,6 @@ private final class TokenStreamCreator: SyntaxVisitor {
   /// token stream.
   func before(_ token: TokenSyntax?, tokens: [Token]) {
     guard let tok = token else { return }
-    for preToken in tokens {
-      if case .open = preToken {
-        openings += 1
-      } else if case .close = preToken {
-        assert(openings > 0)
-        openings -= 1
-      }
-    }
```

Comment:
Had to remove this check and the one on `after`. Do you see any alternative solution so that we can keep these checks? I thought of adding some "unsafe" variants for these special cases, but wanted your feedback first.

Now that I look at it, I'm not even sure how my previous solution didn't fire the assert (at least on the `after(returnTypeToken, tokens: .close)`, since it also didn't have a matching `.open` token)  🤔

Reply:
Ouch! I guess this is fine, since this change is a legitimate place where the `.close` needs to be added to the token stream before the `.open` due to visitation order. If they're imbalanced, it would be detected later by the pretty-printer anyway. Catching that earlier and locally would be nicer, but that's not possible here.

I think the reason your previous solution didn't assert is because the ordering still just happened to keep the total count above zero at all times.

Reply:
Great, thanks!